newIDE/app/src/LoginProvider/BrowserLoginProvider.js

```diff
@@ -46,55 +45,104 @@ class BrowserLoginProvider
 
   async loginOrSignupWithProvider({
     provider,
-    loginOptions,
+    signal,
   }: {|
     provider: IdentityProvider,
-    loginOptions?: ?LoginOptions,
     signal?: AbortSignal,
   |}) {
-    let firebaseProvider = null;
-    if (provider === 'google') {
-      firebaseProvider = new GoogleAuthProvider();
-      firebaseProvider.addScope('profile');
-      firebaseProvider.addScope('email');
-    } else if (provider === 'github') {
-      firebaseProvider = new GithubAuthProvider();
-      // No scope needed for GitHub.
-    } else if (provider === 'apple') {
-      firebaseProvider = new OAuthProvider('apple.com');
-      firebaseProvider.addScope('email');
-      firebaseProvider.addScope('name');
+    if (signal && signal.aborted) {
+      return Promise.reject(
+        new Error('Login or Signup with provider already aborted.')
+      );
     }
-
-    if (!firebaseProvider)
-      throw new Error(`Unknown provider ${provider} for login.`);
-
-    try {
-      await signInWithPopup(this.auth, firebaseProvider);
-      // The user is now stored in `this.auth`.
-      if (loginOptions && loginOptions.notifyConnection) {
-        const { notifyConnection: connectionId } = loginOptions;
-        await this.notifyLogin({ connectionId });
+    const promise = new Promise((resolve, reject) => {
+      // Listen for abort event on signal
+      if (signal) {
+        signal.addEventListener('abort', () => {
+          terminateWebSocket();
+          reject(new Error('Login or Signup with provider aborted.'));
+        });
       }
-    } catch (error) {
-      console.error('Error while login with provider:', error);
-      throw error;
-    }
-  }
+      const width = 400;
+      const height = 600;
+      const left = window.screenX + window.innerWidth / 2 - width / 2;
+      const top = window.screenY + window.innerHeight / 2 - height / 2;
 
-  async notifyLogin({
-    connectionId,
-  }: {|
-    connectionId: string,
-  |}): Promise<void> {
-    const { currentUser } = this.auth;
-    if (!currentUser) return;
+      const authWindow = window.open(
+        'about:blank',
+        null,
+        `width=${width},height=${height},left=${left},top=${top}`
+      );
+      authWindow.document.write(
+        `<style>body { margin: 0; }</style><div style="height: 100vh; display: flex; justify-content: center; align-items: center; color: #F5F5F7; background-color: #25252E; font-family: 'Verdana', 'Fira Sans', 'Open Sans', 'Lucida Sans'; font-size: 20px">Loading</div>`
+      );
+      setupAuthenticationWebSocket({
+        onConnectionEstablished: connectionId => {
+          if (signal && signal.aborted) return;
+          const url = new URL(authenticationPortalUrl);
+          url.searchParams.set('connection-id', connectionId);
+          url.searchParams.set('provider', provider);
+          url.searchParams.set('env', isDev ? 'dev' : 'live');
```

Comment:
Am I understanding correctly that auth.gdevelop.io will be used for both development and production apps for login?

Reply:
Correct - this can be changed later but that's a authentication service and it's more or less always dealing with live data.